src/client.rs

```diff
@@ -14,23 +18,51 @@ use tor_dirmgr::Error;
 use tor_rtcompat::tokio::TokioRustlsRuntime as Runtime;
 use tracing::{debug, trace, warn};
 
+use crate::flatfiledirmgr::check_directory;
 use crate::{
     flatfiledirmgr::FlatFileDirMgrBuilder,
     http::{raw_to_response, request_to_raw},
+    CHURN_FILENAME, MICRODESCRIPTORS_FILENAME,
 };
 
 /// Client using the Tor network
 pub struct Client(TorClient<Runtime>);
+
 /// AUTHORITY_FILENAME is the name of the file containing the authorities.
 pub const AUTHORITY_FILENAME: &str = "authority.json";
 
+#[derive(PartialEq)]
+enum UpdateNeeded {
+    None,
+    Churn,
+    All,
+}
+
+/// Default directory cache download URL, provided by C4DT.
+pub const DIRECTORY_CACHE_C4DT: &str =
+    "https://github.com/c4dt/lightarti-directory/releases/latest/download/directory-cache.tgz";
```

Comment:
while we're at it, can't we make this configurable?

Reply:
Yes, you can call `new_with_url` and give the URL for the directory cache and the directory churn file.

Reply:
Or would it be better if there is only one `new` which takes the two URLs? And then you have to add the URLs also in the example code?

Like:
- `new` takes the URLs
- `new_with_defaults` takes the URLs given for github?